engineio/async_drivers/tornado.py

```diff
@@ -122,12 +122,16 @@ def make_response(status, headers, payload, environ):
     """This function generates an appropriate response object for this async
     mode.
     """
-    tornado_handler = environ['tornado.handler']
-    tornado_handler.set_status(int(status.split()[0]))
-    for header, value in headers:
-        tornado_handler.set_header(header, value)
-    tornado_handler.write(payload)
-    tornado_handler.finish()
+    try:
+        tornado_handler = environ['tornado.handler']
+        tornado_handler.set_status(int(status.split()[0]))
+        for header, value in headers:
+            tornado_handler.set_header(header, value)
+        tornado_handler.write(payload)
+        tornado_handler.finish()
+    except RuntimeError as error:
```

Comment:
What is causing a `RuntimeError` here?

Reply:
Yesterday when I was testing the library with Tornado and Websockets, I saw that the exception is raised when the connection is a websocket (after upgrade) and is closing, Tornado raises the `RuntimeError` using this function. 

```
def _raise_not_supported_for_websockets(*args: Any, **kwargs: Any) -> None:
    raise RuntimeError("Method not supported for Web Sockets")
```

Which is called when the stream is detached. As I could saw, is because we trying to set a method which is not allowed on websocket.
```
def _detach_stream(self) -> IOStream:
        # disable non-WS methods
        for method in [
            "write",
            "redirect",
            "set_header",
            "set_cookie",
            "set_status",
            "flush",
            "finish",
        ]:
            setattr(self, method, _raise_not_supported_for_websockets)
        return self.detach()
```

I did not know if skip `make_response` in case of a websocket will break the compatibility with previous versions, so at least I tried to handle it in the new version.